drivers/SmartThings/matter-lock/src/init.lua

```diff
@@ -30,13 +30,17 @@ local lock_utils = require "lock_utils"
 local YALE_LOCK_FINGERPRINT = {{vendorId = 0x101D, productId = 0x1}}
 
 local function set_cota_credential(device)
-  -- Device requires pin for remote operation if it supports COTA and PIN features.
-  local eps = device:get_endpoints(DoorLock.ID, {feature_bitmap = DoorLock.types.DoorLockFeature.CREDENTIALSOTA | DoorLock.types.DoorLockFeature.PIN_CREDENTIALS})
-  if #eps == 0 then
-    device.log.debug("Device should not require PIN for remote operation, so not setting COTA credential")
+  local eps = device:get_endpoints(DoorLock.ID)
+  local cota_cred = device:get_field(lock_utils.COTA_CRED)
+  if cota_cred == nil then
+    -- Shouldn't happen but defensive to try to figure out if we need the cota cred and set it.
+    device:send(DoorLock.attributes.RequirePINforRemoteOperation:read(device, #eps > 0 and eps[1] or 1))
+    device.thread:call_with_delay(2, function(t) set_cota_credential(device) end)
```

Comment:
Where does the 2 second heuristic come from?

Reply:
This is just a delay to give time for the read to complete before this function is called again. The same sort of delay is used later in this function to wait for the clear credential command to happen before we try to set it.